modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -2294,7 +2294,7 @@ void GDScriptAnalyzer::reduce_call(GDScriptParser::CallNode *p_call, bool p_is_a
 							break;
 #ifdef DEBUG_ENABLED
 						} else {
-							if (par_type.builtin_type == Variant::INT && p_call->arguments[i]->get_datatype().builtin_type == Variant::FLOAT) {
+							if (par_type.builtin_type == Variant::INT && p_call->arguments[i]->get_datatype().builtin_type == Variant::FLOAT && p_call->function_name != SNAME("int")) {
```

Comment:
I think it's better (and probably faster) to check the builtin type, since that was already solved before.
```suggestion
							if (par_type.builtin_type == Variant::INT && p_call->arguments[i]->get_datatype().builtin_type == Variant::FLOAT && builtin_type != Variant::INT) {
```

Reply:
Thanks! Done